k4FWCore/components/Writer.cpp

```diff
@@ -263,6 +266,9 @@ class Writer final : public Gaudi::Functional::Consumer<void(const EventContext&
 
     debug() << "Writing frame" << endmsg;
     iosvc->getWriter().writeFrame(ptr->getData(), podio::Category::Event, m_collectionsToSave);
+    if (deletePtr && m_collectionsToAdd.empty()) {
```

Comment:
Why is the condition here also checking the `m_collectionsToAdd` to be empty?

Reply:
Outdated